### PR TITLE
TGW Route Additions

### PR DESCRIPTION
**Pull request by torr201812**

### CODE REVIEW

**Comment by userhas404d** [comment]
@torr201812 looks like the the documentation linting make target isn't happy. A `make docs/generate` should fix it though

**Comment by userhas404d** [comment]
also, could you add a test for these please?

**Comment by lorengordon** [comment]
Ok, I've updated this to create/manage a list of routes with the tgw as the destination. I ended up using `count` instead of `for_each` for the route resources. That was because I could not find a way to specify a list/map of route objects that made sense and avoided failures when the route tables were created in the same state (`for_each` depending on resources not yet existing).

The risk using count is that if the order of the list changes, then tf will want to destroy/create resources. I think this is safe in this case though, as we've been using count in the pcx module for a while now.

**Comment by lorengordon** [comment]
Results of `make test`:

```
PASS
ok      terraform-aws-tardigrade-transit-gateway-attachment/tests       463.575s
```

File: tests/create_attachment/prereq/main.tf [hunk 1]
```diff
@@ -8,11 +8,15 @@ provider aws {
 
 provider aws {
   region  = "us-east-1"
-  profile = "owner"
   alias   = "owner"
+  profile = "owner"
```
Comment (confusdcodr) [inline tests/create_attachment/prereq/main.tf]: Please add to the documentation that a user who is looking to test the module will need to have a profile called `owner`

Comment (lorengordon) [reply tests/create_attachment/prereq/main.tf]: That was already in the readme, but I went ahead and made a tiny adjustment to the formatting.

**Pull request merged**